### PR TITLE
Fix group by in DAGRequest

### PR DESCRIPTION
**Pull request by Novemser**
Add a `First` aggregate function to stay in accordance with TiDB's implementation when building DAG request.

### CODE REVIEW

**Comment by Novemser** [comment]
/run-all-tests tikv=release-1.1 tidb=release-1.1 pd=release-1.1

**Comment by Novemser** [comment]
Refer to https://github.com/pingcap/tispark/issues/255

**Comment by Novemser** [comment]
/run-all-tests tikv=release-1.1 tidb=release-1.1 pd=release-1.1

File: core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala [hunk 1]
```diff
@@ -17,6 +17,32 @@ package org.apache.spark.sql
 
 class IssueTestSuite extends BaseTiSparkSuite {
 
+  // https://github.com/pingcap/tispark/issues/255
+  test("Group by with first") {
+    ti.tidbMapDatabase("tpch_test")
+    val q1 = spark.sql("""
```
Comment (birdstorm) [inline core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala]: may be better to change this test to `run(sql)` form.

Comment (Novemser) [reply core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala]: Done

**Pull request merged**